lib/search/presenters/spell_check_presenter.rb

```diff
@@ -3,13 +3,27 @@ module Search
     def present
       return [] unless any_suggestions?
 
-      es_response["suggest"]["spelling_suggestions"].first["options"].map { |option| option["text"] }
+      suggestions.map do |option|
+        if highlighted_suggestions?
+          { text: option["text"], highlighted: option["highlighted"] }
```

Comment:
Is this logic so we AB test this?

Reply:
It's so that I don't need to do something like `if suggestion.is_a? String` in finder-frontend, as a change will also be needed for that.  When finder-frontend is deployed this can become the default behaviour and the `spelling_with_highlighting` removed.